subgraphs/uniswap-v2/src/core.ts

```diff
@@ -0,0 +1,527 @@
+  if (event.params.to.toHexString() == pool.id) {
+    let burns = transaction.burns
+    let length = burns == null ? 0: burns.length
+    let burn = new BurnEvent(
+      event.transaction.hash
+        .toHexString()
+        .concat('-')
+        .concat(BigInt.fromI32(length).toString())
+    )
+
+    burn.liquidity = value
+    burn.to = event.params.to
+    burn.sender = event.params.from
+    burn.needsComplete = true
+    burn.save()
+
+    if (burns == null) transaction.burns = [burn.id]
+    else transaction.burns = burns.concat([burn.id])
+
+    transaction.save()
+  }
+
+  // burn
+  if (event.params.to.toHexString() == ZERO_ADDRESS && event.params.from.toHexString() == pool.id) {
+    // Tracks supply of minted LP tokens 
+    pool.outputTokenSupply = pool.outputTokenSupply.minus(value)
+    pool.save()
+
+    // this is a new instance of a logical burn
+    let burns = transaction.burns
+    let burn: BurnEvent
+    if (burns != null) {
+      let currentBurn = BurnEvent.load(burns[burns.length - 1])
+      if (currentBurn != null && currentBurn.needsComplete) {
+        burn = currentBurn as BurnEvent
+      } else {
+        burn = new BurnEvent(
+          event.transaction.hash
+            .toHexString()
+            .concat('-')
+            .concat(BigInt.fromI32(burns.length).toString())
+        )
+
+        burn.needsComplete = false
+        burn.liquidity = value
+        burn.to = event.params.to
+        burn.sender = event.params.from
+      }
+    } else {
+      burn = new BurnEvent(
+        event.transaction.hash
+          .toHexString()
+          .concat('-')
+          .concat(BigInt.fromI32(0).toString())
+      )
+
+      burn.needsComplete = false
+      burn.liquidity = value
+      burn.to = event.params.to
+      burn.sender = event.params.from
+
+    }
+
+    // if this logical burn included a fee mint, account for this
+    if (mints != null && !isCompleteMint(mints[mints.length - 1])) {
+      // remove the logical mint
+      store.remove('_Mint', mints[mints.length - 1])
+      // update the transaction
+
+      // TODO: Consider using .slice().pop() to protect against unintended
+      // side effects for other code paths.
+      mints.pop()
+      transaction.mints = mints
+      transaction.save()
+    }
+    burn.save()
+    // if accessing last one, replace it
+    if (burn.needsComplete) {
+      // TODO: Consider using .slice(0, -1).conc at() to protect against
+      // unintended side effects for other code paths.
+
+      if (burns != null) burns[burns.length - 1] = burn.id
+      else burns = [burn.id]
+    }
+    // else add new one
+    else {
+      if (burns == null) transaction.burns = [burn.id]
+      else transaction.burns = burns.concat([burn.id])
+    }
+    transaction.burns = burns
+    transaction.save()
+  }
+
+  transaction.save()
+}
+
+export function handleSync(event: Sync): void {
+  let protocol = DexAmmProtocol.load(FACTORY_ADDRESS)
+  if (protocol == null) return
```

Comment:
You should probably use the `getOrCreateXXX` pattern here and in other places as well instead of `load`

Reply:
Updated to this pattern. Used widely now for most/all entities.